src/blocktree_processor.rs

```diff
@@ -189,12 +192,34 @@ where
 }
 
 pub fn process_blocktree(
-    bank_forks: &BankForks,
+    genesis_block: &GenesisBlock,
     blocktree: &Blocktree,
     leader_scheduler: &Arc<RwLock<LeaderScheduler>>,
-) -> Result<(u64, Hash)> {
+) -> Result<(BankForks, u64, Hash)> {
+    let bank = Bank::new(&genesis_block);
+    let slot_height = 0; // Use the Bank's slot_height as its ID.
+    let bank_forks = BankForks::new(slot_height, bank);
+    leader_scheduler
+        .write()
+        .unwrap()
+        .update_tick_height(0, &bank_forks.finalized_bank());
+
+    let now = Instant::now();
+    info!("processing ledger...");
     let entries = blocktree.read_ledger().expect("opening ledger");
-    process_ledger(&bank_forks.working_bank(), entries, leader_scheduler)
+    let (entry_height, last_entry_id) =
+        process_ledger(&bank_forks.working_bank(), entries, leader_scheduler)?;
+
+    info!(
+        "processed {} ledger entries in {}ms, tick_height={}...",
+        entry_height,
+        duration_as_ms(&now.elapsed()),
+        bank_forks.working_bank().tick_height()
+    );
+
+    // TODO: probably need to return `entry_height` and `last_entry_id` for *all* banks in
```

Comment:
Just don't return it. It's available via `bank_forks.working_bank().tick_height()`.

Reply:
Sadly no.  `entry_height` is not known to the Bank, it's a ledger-only construct